test/unit/transaction.test.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+/* jshint -W030 */
+var chai = require('chai')
+, expect = chai.expect
+, sinon = require('sinon')
+, Support = require(__dirname + '/support')
+, Sequelize = Support.Sequelize
+, dialect = Support.getTestDialect()
+, current = Support.sequelize
+, Promise = Sequelize.Promise;
+
+describe('Transaction', function() {
+
+  before(function () {
+    this.stub = sinon.stub(current, 'query').returns(Promise.resolve({}));
+
+    this.stubConnection = sinon.stub(current.connectionManager, 'getConnection')
+    .returns(Promise.resolve({ uuid: 'ssfdjd-434fd-43dfg23-2d', close : function() { }}));
+  });
+
+  beforeEach(function () {
+    this.stub.reset();
+    this.stubConnection.reset();
+  });
+
+  after(function () {
+    this.stub.restore();
+    this.stubConnection.restore();
+  });
+
+  it('should run auto commit query only when needed', function() {
+    var expectations = {
+      all: [
+        'START TRANSACTION;'
+      ],
+      mysql: [
+        'START TRANSACTION;',
+        'SET autocommit = 1;'
```

Comment:
@sushantdhiman Actually, I'm not sure why this is needed? As I read the mysql docs 
> with autocommit mode enabled. This means that as soon as you execute a statement that updates (modifies) a table, MySQL stores the update on disk to make it permanent. The change cannot be rolled back.

This is the normal mode, outside of transactions.


>To disable autocommit mode implicitly for a single series of statements, use the START TRANSACTION statement:

http://dev.mysql.com/doc/refman/5.7/en/commit.html

So disabling autocommit with start transaction, then immediately enabling it again feels wrong.